client-app/app/controllers/index.js

```diff
@@ -171,4 +182,60 @@ export default class IndexController extends Controller {
 
     debounce(this, this.doSearch, term, 250);
   }
+
+  @action
+  buildGroupingRegexFromSelectedRows() {
+    this.buildingGroupingRegex = true;
+  }
+
+  @action
+  async createGroupingRegexFromSelectedRows() {
+    if (this.rowMessagesForGroupingRegex.length < 2) {
+      // eslint-disable-next-line no-alert
+      return alert(
+        "You must select at least 2 rows to create a grouping regex"
+      );
+    } else {
+      const match = this.findLongestMatchingSubstring(
+        this.rowMessagesForGroupingRegex
+      );
+
+      if (
+        // eslint-disable-next-line no-alert
+        confirm(
+          `Do you want to create the pattern\n\n"${match}"\n\nCancel = No, OK = Create`
+        )
+      ) {
+        await ajax("/patterns/grouping.json", {
+          method: "POST",
+          data: {
+            pattern: match,
+          },
+        });
+        this.rowMessagesForGroupingRegex = [];
+        this.buildingGroupingRegex = false;
+        this.model.reload();
+      }
+    }
+  }
+
+  findLongestMatchingSubstring(strings) {
```

Comment:
This just checks the beginning of the string, right? If so I think it would be better named `findLongestMatchingPrefix`. Otherwise it gives the impression it might find matches in the middle/end of strings as well.